authzed/api/v1/acl_service.proto

```diff
@@ -153,42 +180,60 @@ message CheckPermissionResponse {
   }
 
   ZedToken checked_at = 1;
+
+  // permissionship is the result of the check operation. For a permission, this will be
+  // PERMISSIONSHIP_HAS_PERMISSION if the subject has said permission; for a relation, this will
+  // be PERMISSIONSHIP_HAS_PERMISSION if the subject is a member of the relation via the existance
+  // of a relationship.
   Permissionship permissionship = 2;
 }
 
+// ExpandPermissionTreeRequest returns a tree representing the expansion of all relationships
+// found accessible from a permission or relation on a particular resource.
+//
+// ExpandPermissionTreeRequest is typically used to determine the full set of subjects with
+// a permission, along with the relationships that grant said access.
 message ExpandPermissionTreeRequest {
-  // The Expand API returns the effective userset given an resource#relation⟩
-  // pair and an optional ZedToken. Unlike the Read API, Expand follows indirect
-  // references expressed through userset rewrite rules.
   Consistency consistency = 1;
+
+  // resource is the resource over which to run the expansion.
   ObjectReference resource = 2 [ (validate.rules).message.required = true ];
+
+  // permission is the name of the permission or relation over which to run the expansion for
+  // the resource.
   string permission = 3 [ (validate.rules).string = {
     pattern : "^([a-z][a-z0-9_]{2,62}[a-z0-9])?$",
     max_bytes : 64,
   } ];
 }
 
 message ExpandPermissionTreeResponse {
-  // The result is represented by a userset tree whose leaf nodes are user IDs
-  // or usersets pointing to other resource#relation⟩ pairs, and intermediate
-  // nodes represent union, intersection, or exclusion operators.
   ZedToken expanded_at = 1;
+
+  // tree_root is a tree structure whose leaf nodes are subjects, and intermediate
+  // nodes represent the various operations (union, intersection, exclusion) to reach
+  // those subjects.
   PermissionRelationshipTree tree_root = 2;
 }
 
+// LookupResourcesRequest performs a reverse-lookup of all resources of a particular kind on
```

Comment:
I'm not sure mentioning the the "reversal" aspect of lookup does anything but confuse the reader.

>LookupResourcesRequest discovers all resources of a particular kind on which the subject has the specified permission or the relation in which the subject exists.

Reply:
Changed